pkg/gui/commit_files_panel.go

```diff
@@ -45,6 +45,7 @@ func (gui *Gui) SwitchToCommitFilesContext(opts controllers.SwitchToCommitFilesC
 
 	gui.State.Contexts.CommitFiles.SetSelectedLineIdx(0)
 	gui.State.Contexts.CommitFiles.SetRefName(opts.RefName)
+	gui.State.Contexts.CommitFiles.SetParentRefName(opts.ParentRefName)
```

Comment:
Managing both a RefName and ParentRefName field is a little cumbersome. What are your thoughts on having a new interface called Ref which has a RefName and ParentRefName method and we can go `CommitFiles.SetRef` and `CommitFiles.GetRef`. 

It also means that in our contexts we can have a GetSelectedRef method instead of GetSelectedRef and GetParentRef methods, where we can go:
```go
func (self *LocalCommitsContext) GetSelectedRef() Ref {
  item := context.GetSelected()
  if item == nil { // important to do this so that we return `nil` with the right type
    return nil
  }
  return item
}
```
If Go had covariance we wouldn't need the extra ceremony but alas